### PR TITLE
story(ccmspui-314) general requests document upload screens

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
includes:
- ability to add new documents to a general request
- ability to delete documents
- maintains form data between pages
- form validation
- custom js, copied from pui for large description columns

### CODE REVIEW

**Comment by PhilDigitalJustice** [comment]
Unit test to amend on return from leave

**Comment by PhilDigitalJustice** [comment]
Unit tests finished. Amended CSS/JS to new format.

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/section/EvidenceSectionController.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public String viewEvidenceUpload(
 
     // Initialise the form data object.
     final EvidenceUploadFormData evidenceUploadFormData = new EvidenceUploadFormData();
-    evidenceUploadFormData.setApplicationOrOutcomeId(activeCase.getApplicationId());
+    evidenceUploadFormData.setApplicationOrOutcomeId(activeCase.getApplicationId().toString());
```
Comment (JamieBriggs-MoJ) [inline src/main/java/uk/gov/laa/ccms/caab/controller/application/section/EvidenceSectionController.java:131]: Could do `String.valueOf(activeCase.getApplicationId())` to make this line null safe

**Review by JamieBriggs-MoJ — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/laa/ccms/caab/bean/request/ProviderRequestFlowFormData.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package uk.gov.laa.ccms.caab.bean.request;
 
+import java.util.UUID;
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/bean/request/ProviderRequestFlowFormData.java]: updated

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/section/EvidenceSectionController.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/application/section/EvidenceSectionController.java]: Updated

**Comment by PhilDigitalJustice** [comment]
All comments addressed

**Review by JamieBriggs-MoJ — APPROVED** [review]
👍🏻

**Review by farrell-m — APPROVED** [review]
👍

**Pull request merged**